core/src/main/scala/zio/config/ConfigDocsFunctions.scala

```diff
@@ -63,31 +60,39 @@ private[config] trait ConfigDocsFunctions {
   ): Either[String, ConfigDocs[K, V]] =
     write(config, value)
       .map(tree => {
-        val flattened = tree.flatten
-
-        def loop(c: ConfigDocs[K, V], initialValue: Vector[K]): ConfigDocs[K, V] =
+        def loop(
+          tree: PropertyTree[K, V],
+          flattened: Map[Vector[K], ::[V]],
+          c: ConfigDocs[K, V],
+          initialValue: Vector[K]
+        ): ConfigDocs[K, V] =
           c match {
             case Empty => Empty
 
-            case Path(path, docs) =>
-              val updated: Details[V] =
-                docs match {
-                  case Descriptions(sources, descriptions) =>
-                    DescriptionsWithValue(flattened.get(initialValue :+ path).map(_.head), sources, descriptions)
-                  case a => a
-                }
-              Path(path, updated)
+            case Descriptions(sources, descriptions) =>
+              DescriptionsWithValue(flattened.get(initialValue).map(_.head), sources, descriptions)
+
+            case a: DescriptionsWithValue[V] => a
 
             case NestedPath(path, docs) =>
-              NestedPath(path, loop(docs, initialValue :+ path))
+              NestedPath(path, loop(tree, flattened, docs, initialValue :+ path))
 
             case Both(left, right) =>
-              Both(loop(left, initialValue), loop(right, initialValue))
+              Both(loop(tree, flattened, left, initialValue), loop(tree, flattened, right, initialValue))
 
             case OneOf(left, right) =>
-              OneOf(loop(left, initialValue), loop(right, initialValue))
+              OneOf(loop(tree, flattened, left, initialValue), loop(tree, flattened, right, initialValue))
+
+            case Sequence(element) =>
+              tree.getPath(initialValue.toList) match {
+                case PropertyTree.Sequence(value) if value.nonEmpty =>
+                  SequenceWithValue(value.map(t => loop(t, t.flatten, element, Vector.empty)))
```

Comment:
With the new Details, I think, `SequenceWithValue` can be removed, and have Sequence as `Sequence(list: List[ConfigDocs])` and provide a Sequence(List(...)) if the loop produces only one entry.

Reply:
done